geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ExportLogsInterceptor.java

```diff
@@ -106,8 +159,78 @@ public ResultModel postExecution(GfshParseResult parseResult, ResultModel comman
     }
 
     // if there is no downloaded file. File is saved on the locator/manager.
-    return ResultModel.createInfo(
-        "Logs exported to the connected member's file system: "
-            + commandResult.getFileToDownload().toString());
+    output.append("Logs exported to the connected member's file system: ")
+        .append(commandResult.getFileToDownload().toString());
+    return ResultModel.createInfo(output.toString());
+  }
+
+  private ResultModel checkStartAndEndFormat(String start, String end) {
+    StringBuilder formatErrorMessage = new StringBuilder();
+    boolean formatError = false;
+    if (start != null && !DATE_AND_TIME_PATTERN.matcher(start).matches()
+        && !DATE_ONLY_PATTERN.matcher(start).matches()) {
+      formatErrorMessage.append(EXPORT_LOGS__STARTTIME);
+      formatError = true;
+    }
+
+    if (end != null && !DATE_AND_TIME_PATTERN.matcher(end).matches()
+        && !DATE_ONLY_PATTERN.matcher(end).matches()) {
+      if (formatError) {
+        formatErrorMessage.append(" and ");
+      }
+      formatErrorMessage.append(EXPORT_LOGS__ENDTIME);
+      formatError = true;
+    }
+
+    if (formatError) {
+      formatErrorMessage.append(" had incorrect format. Valid formats are ")
+          .append(ONLY_DATE_FORMAT).append(" and ").append(FORMAT);
+      return ResultModel.createError(formatErrorMessage.toString());
+    }
+    return null;
+  }
+
+  private ResultModel checkStartAndEndParsing(String start, String end) {
+    StringBuilder parseErrorMessage = new StringBuilder();
+    boolean parseError = false;
+
+    SimpleDateFormat dateAndTimeFormat = new SimpleDateFormat(FORMAT);
+    SimpleDateFormat dateOnlyFormat = new SimpleDateFormat(ONLY_DATE_FORMAT);
+    if (start != null) {
+      try {
+        // If the input is intended to be parsed as date and time, use the date and time format
+        if (start.length() > 10) {
+          dateAndTimeFormat.parse(start);
+        } else {
+          dateOnlyFormat.parse(start);
+        }
+      } catch (ParseException e) {
```

Comment:
Might be a nitpick. Is it possible to append the ParseException error message as well?

Reply:
The error message from the ParseException is simply "Unparseable date: " followed by the user input, so it doesn't add any useful information here.